docs/image-manipulations/optimizing-images.md

```diff
@@ -28,17 +28,26 @@ No matter where or how many times you call `optimize` in you chain, it will alwa
 
 ## Customizing the optimization
 
-To optimization of images is done by the underlying [spatie/image-optimizer](https://github.com/spatie/image-optimizer) package. You can pass your own customized chains as array. The keys should be fully qualified class names of optimizers and the values the options that they should get. Here's an example
+You are able to customize the optimization by passing an options array to the `optimize` method:
+- `'timeout'` settings key lets you to set a custom timeout in seconds other than the default 60s. Adjusting this setting may be inevitable while working with large images (see e.g. [#187](https://github.com/spatie/image/pull/187))
+- `'optimizers'` settings key lets you to pass your own customized chain of optimizers as an array. The keys should be fully qualified class names of optimizers and the values should be the options that they should get
+
+Here's an example:
 
 ```php
 Image::load('example.jpg')
-    ->optimize([Jpegoptim::class => [
-        '--all-progressive',
-    ]])
+    ->optimize([
+        'timeout' => 120,
```

Comment:
instead of adding a key in the array, let's add a dedicated parameter to specify timeout.

So this would use default timeout.

```
php
Image::load('example.jpg')
     ->optimize([
         'optimizers' => [
             Jpegoptim::class => [
                 '--all-progressive',
             ],
         ],
     ])
```

and this is how you would use a custom timeout:


```php
Image::load('example.jpg')
     ->optimize([
         'optimizers' => [
             Jpegoptim::class => [
                 '--all-progressive',
             ],
         ],
     ], timeout: 120)
```

Reply:
Hello @freekmurze :wave: Thanks for your feedback! I've updated the PR adding the `$timeout` parameter.

It seems that the only option I had to implement this was to reuse that array key. It's being added under the hood when the `$timeout` parameter is being passed:
```php
    if ($timeout !== null) {
        $optimizationOptions['timeout'] = $timeout;
    }
```
The optimization options are always passed to the underlying layers with JSON serialization/deserialization:
```php
    return $this->addManipulation('optimize', json_encode($optimizationOptions));
```
That's why only a single array is available for any configuration.

Moreover, in a Laravel project with [the `spatie/laravel-medialibrary`](https://spatie.be/docs/laravel-medialibrary/v10/introduction) one might want to be able to configure a timeout for optimizers. This can be achieved with the help of the [`image_optimizers` settings key](https://github.com/spatie/laravel-medialibrary/blob/75087ebbd5c80445edf6a6736f2685dbf87859e5/config/media-library.php#L113) in the `config/media-library.php` adding e.g. `'timeout' => 120` to the array:
```php
    'image_optimizers' => [
        'optimizers' => [
            Spatie\ImageOptimizer\Optimizers\Jpegoptim::class => [
                '-m85',
                // ...
            ],
        ],
        'timeout' => 120,
    ],
```

So I think the only possible way is to have both options available.